lib/publiccloud/ssh_interactive.pm

```diff
@@ -1,67 +1,98 @@
 # SUSE's openQA tests
 #
-# Copyright 2019 SUSE LLC
+# Copyright 2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 
 # Summary: Class with helpers related to SSH Interactive mode
 #
-# Maintainer: Pavel Dostal <pdostal@suse.cz>
+# Maintainer: qa-c@suse.de
 
 package publiccloud::ssh_interactive;
+use testapi qw(is_serial_terminal);
 use base Exporter;
 use testapi;
 use Utils::Backends qw(set_sshserial_dev unset_sshserial_dev);
+use utils qw(type_string_very_slow);
 use strict;
 use warnings;
+use distribution;
 
 our @EXPORT = qw(ssh_interactive_tunnel ssh_interactive_leave select_host_console);
 
+sub establish_tunnel_console {
+    # Helper call to activate a console and establish the ssh connection therein
+    my $console = $_[0];
+
+    select_console("$console");
+    # Note: Don't use script_run here! The serial terminal is set to /dev/sshserial, so every script_run will time out
+    type_string("\n~.\n", max_interval => 1);    # ensure no previous ssh connection is present
+    enter_cmd("clear");
+    enter_cmd('ssh -t sut');
+    # give the ssh connection some time to settle
+    sleep 5;
```

Comment:
maybe from the beginning we will move this into var ? Something telling me that move than one commit in future will tweak this line :)

Reply:
How about we remove it? You may assert the prompt from the remote server instead.

Reply:
Or use something like `script_retry('true');` in order to wait until you get your return code which means that you're logged in the instance.

Reply:
We don't get anything from doing so. `sleep` itself is not evil. One should not use sleep for synchronization problems but that's not the case here.

The intention is to give ssh some time to establish the connection before proceeding. It takes typically 1-2 seconds, so a 5 second sleep here is more than enough for most typical cases.

Also: We can't use any `script_run`, `script_retry`, `assert_script_run` or anything at this point because the tunnel is not yet established.

Reply:
the trap here is that you sleeping on worker to wait for something which happening inside VM . And I can say you with big confidence as witness of dozens of similar stories when initialization which in normal circumstances takes 1-2 seconds can lasts up to one minute on super overloaded worker . 
BUT I agree to deal "till first tweak" :)

Reply:
There is also `ssh -q -o BatchMode=yes sut` which can query the machine and check if it is up or not. would that work if you wrap it with `script_retry`?

Reply:
Again: We can't use any `script_run`, `script_retry`, `assert_script_run` or anything at this point because the tunnel is not yet established.